clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -404,6 +411,12 @@ private void maybeTriggerWakeup() {
             throw new WakeupException();
         }
     }
+    
+    private void maybeThrowInterruptException() {
```

Comment:
Thanks, I think it's reasonable to make the behavior of the two a little different. It basically gives you a hard kill option, which we don't have at the moment. By the way, it looks like we catch exceptions raised from `KafkaConsumer.close()` and rethrow them as a `KafkaException`. I guess we should probably throw the `InterruptException` instead?